packages/genesys-spark-components/src/components/beta/gux-avatar/gux-avatar.tsx

```diff
@@ -0,0 +1,173 @@
+      return nameArray[0].charAt(0) + nameArray[nameArray.length - 1].charAt(0);
+    }
+    return nameArray[0]?.charAt(0) + nameArray[0]?.charAt(1);
+  }
+
+  private displayInnerContent(): JSX.Element | null {
+    if (this.hasImageSlot) {
+      return (<slot name="image"></slot>) as JSX.Element;
+    }
+    return (
+      <div class="initials">{this.generateInitials()}</div>
+    ) as JSX.Element;
+  }
+
+  private displayBadge(): JSX.Element | null {
+    const displayedStatus = this.notifications ? 'notifications' : this.status;
+    if (this.badge) {
+      return (
+        <div
+          class={{
+            'gux-avatar-badge': true,
+            [`${displayedStatus}`]: true,
+            [`gux-${this.size}`]: true
+          }}
+        >
+          <gux-icon
+            icon-name={this.getIcon(displayedStatus)}
+            screenreader-text={displayedStatus}
+          ></gux-icon>
+        </div>
+      ) as JSX.Element;
+    }
+  }
+
+  private getAccent(): string {
+    if (this.accent === 'auto') {
+      let hash = 0;
+      for (let i = 0; i < this.name.length; i++) {
+        hash += this.name.charCodeAt(i);
+      }
+      const accent = hash % 12;
+      return accent === 0 ? '12' : accent.toString();
+    }
+    return this.accent.toString();
+  }
+
+  private getIcon(status: GuxAvatarStatus): string {
+    switch (status) {
+      case 'available':
+        return 'fa/circle-check-solid';
+      case 'break':
+        return 'fa/clock-outline';
+      case 'busy':
+        return 'fa/ban-outline';
+      case 'notifications':
+        return 'fa/bell-regular';
+      case 'offline':
+        return 'fa/circle-xmark-regular';
+      case 'out-of-office':
+        return 'fa/subtract-circle';
+      case 'queue':
+        return 'fa/headset-solid';
+      default:
+        return 'fa/circle-xmark-regular';
+    }
+  }
+
+  componentWillLoad(): void {
+    trackComponent(this.root, { variant: this.status });
+    this.hasImageSlot = hasSlot(this.root, 'image');
+  }
+
+  componentDidLoad() {
+    if (!this.name) {
+      logWarn(this.root, 'must have a name attribute for accessibility.');
+    }
+  }
+
+  render(): JSX.Element {
+    const avatarContent = (
+      <Fragment>
+        <div
+          class={{
+            'gux-avatar': true,
+            [`${this.status}`]: true,
+            [`gux-${this.size}`]: true,
+            [`gux-${this.statusRing ? 'status-ring' : ''}`]: this.statusRing
+          }}
+          aria-label={this.name}
+        >
+          <div
+            class={{
+              content: true,
+              [`gux-accent-${this.getAccent()}`]: true
+            }}
+            aria-hidden="true"
+          >
+            {this.displayInnerContent()}
+          </div>
+        </div>
+        {this.displayBadge()}
+      </Fragment>
+    );
+
+    return this.interactive
+      ? ((<button>{avatarContent}</button>) as JSX.Element)
+      : (avatarContent as JSX.Element);
```

Comment:
I would maybe do this logic in a function outside the render function and call it in here.